### DISCUSSION TITLE
Reduce config load and increase priority

### DISCUSSION THREAD

**Discussion by vvanglro** [Kludex/uvicorn#2440] [Q&A] [answered]
In the new process manager mode, if I pass in the wrong app, the process keeps restarting.

After I read the `config.load()` call logic of the entire project code, I think I can reduce unnecessary calls in the code and increase the priority, that is, continue to create sockets and process management after the load is completed.

@Kludex What do you think?

**Comment by Kludex** [comment, marked as answer]
If you can show me some code, it would be nice.

**Reply by vvanglro** [reply]
https://github.com/encode/uvicorn/pull/2443

**Comment by vvanglro** [comment]
```py

# uvicorn_test.py

from fastapi import FastAPI

app = FastAPI()


@app.get("/")
async def root():
    return {"message": "Hello World"}


if __name__ == '__main__':
    import uvicorn
    uvicorn.run("uvicorn_test:app2", host="0.0.0.0", port=8000, workers=3)
```
In multiprocess mode this will result in constant reboots because sys.exit occurs in a child process. https://github.com/encode/uvicorn/blob/master/uvicorn/config.py#L435-L439

So if we put the loading app in the main process, this will be solved. https://github.com/encode/uvicorn/pull/2444
